transform-plugins/src/main/java/io/cdap/plugin/XMLParser.java

```diff
@@ -316,6 +344,25 @@ public Config(String inputField, String encoding, String xPathFieldMapping, Stri
       this.fieldTypeMapping = fieldTypeMapping;
       this.processOnError = processOnError;
       this.failOnArray = false;
+      this.enableExternalGeneralEntities = false;
+      this.enableExternalParameterEntities = false;
+      this.loadExternalDTD = false;
+      this.disallowDocTypeDTD = true;
```

Comment:
the default in the backend should be false for backwards compatibility. 

The default in the widget can be true if we think that's better a better default moving forward.

Reply:
+1, would prefer the default in the widget be false. It should be fine to have an empty plugin leave external entities enabled, but we should consider that copying pipeline JSONs will persist this "vulnerability".